Source/Scene/LabelCollection.js

```diff
@@ -213,46 +213,88 @@ define([
+                lineWidth = 0;
+            }
         }
 
         var scale = label._scale;
         var horizontalOrigin = label._horizontalOrigin;
         var widthOffset = 0;
         if (horizontalOrigin === HorizontalOrigin.CENTER) {
-            widthOffset -= totalWidth / 2 * scale;
+            //widthOffset -= totalWidth / 2 * scale;
+            widthOffset -= maxWidth / 2 * scale;
         } else if (horizontalOrigin === HorizontalOrigin.RIGHT) {
-            widthOffset -= totalWidth * scale;
+            //widthOffset -= totalWidth * scale;
+            widthOffset -= maxWidth * scale;
+        }
+        var heightOffset = 0;
+        var totalHeight = maxHeight * numberNewlines;
+        var verticalOrigin = label._verticalOrigin;
+        if (verticalOrigin === VerticalOrigin.CENTER) {
+            heightOffset -= totalHeight / 2 * scale;
+        } else if (verticalOrigin === VerticalOrigin.TOP) {
+            heightOffset -= totalHeight * scale;
         }
+        else if (verticalOrigin === VerticalOrigin.BOTTOM) {
+            heightOffset += totalHeight * scale;
+         }
 
         glyphPixelOffset.x = widthOffset * resolutionScale;
         glyphPixelOffset.y = 0;
 
-        var verticalOrigin = label._verticalOrigin;
+        //var verticalOrigin = label._verticalOrigin;
+        var glyphNewlineOffset = 0;
         for (glyphIndex = 0; glyphIndex < glyphLength; ++glyphIndex) {
             glyph = glyphs[glyphIndex];
             dimensions = glyph.dimensions;
 
             if (verticalOrigin === VerticalOrigin.BOTTOM || dimensions.height === maxHeight) {
-                glyphPixelOffset.y = -dimensions.descent * scale;
+                //glyphPixelOffset.y = -dimensions.descent * scale;
+                glyphPixelOffset.y = heightOffset - dimensions.descent * scale;
             } else if (verticalOrigin === VerticalOrigin.TOP) {
-                glyphPixelOffset.y = -(maxHeight - dimensions.height) * scale - dimensions.descent * scale;
+               // glyphPixelOffset.y = -(maxHeight - dimensions.height) * scale - dimensions.descent * scale;
+               glyphPixelOffset.y = heightOffset - (maxHeight - dimensions.height) * scale - dimensions.descent * scale;
             } else if (verticalOrigin === VerticalOrigin.CENTER) {
-                glyphPixelOffset.y = -(maxHeight - dimensions.height) / 2 * scale - dimensions.descent * scale;
+                //glyphPixelOffset.y = -(maxHeight - dimensions.height) / 2 * scale - dimensions.descent * scale;
+               glyphPixelOffset.y = heightOffset - (maxHeight - dimensions.height) / 2 * scale - dimensions.descent * scale;
+            }
+            if (text.charAt(glyphIndex) === '\n') {
+                glyphNewlineOffset += maxHeight * scale;
+                glyphPixelOffset.x = widthOffset;
+                continue;
+             }
+            if (verticalOrigin === VerticalOrigin.TOP){
+                glyphPixelOffset.y -= glyphNewlineOffset;
```

Comment:
Since the final else does the same thing as this block, you should be able to just remove it.  Also, please clean up the formatting here also.